### DISCUSSION TITLE
Tasmota Lags badly when not connected to Internet using DS3231 for RTC

### DISCUSSION THREAD

**Discussion by brunofosi** [arendst/Tasmota#12855] [Q&A] [answered]
I'm trying to setup a few ESP8266 NodeMCU on a closed ethernet without internet access! ...And for that i've used a RTC as stated here: https://tasmota.github.io/docs/DS3231/ so I don't need to use an extra local NTP server.

**1º** [enable the DS3231 support, compile and upload it to the ESP8266 ](https://tasmota.github.io/docs/DS3231/)
**2º** setup a generic Tasmota with internet available to catch time and reboot to see if everything is ok:

> 00:00:00.002 HDW: ESP8266EX
> 00:00:00.050 CFG: Loaded from flash at F4, Count 12
> 00:00:00.056 QPC: Count 1
> 00:00:00.064 Project tasmota - Tasmota Version 9.5.0.5(tasmota)-2_7_4_9(2021-08-09T21:38:59)
> 00:00:00.066 I2C: DS3231 found at 0x68
> 00:00:00.076 Set time from DS3231 to RTC (UTC) 2021-08-09T20:51:52, (DST) 2021-03-28T02:00:00, (STD) 2021-10-31T03:00:00
> 00:00:00.554 WIF: Connecting to AP1 xxxxx Channel 6 BSSId xxxxxxxx in mode 11n as tasmota-xxxxxx-xxxxx...
> 21:51:53.753 WIF: Connected
> 21:51:54.004 HTP: Web server active on tasmota-xxxxxx-xxxx with IP address 192.168.2.xx
> 21:52:05.015 RSL: INFO1 = {"Info1":{"Module":"Generic","Version":"9.5.0.5(tasmota)","FallbackTopic":"cmnd/DVES_xxxxxx_fb/","GroupTopic":"cmnd/tasmotas/"}}
> 21:52:05.019 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-xxxxxx-7913","IPAddress":"192.168.2.xx"}}
> 21:52:05.030 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart"}}
> 21:52:08.422 QPC: Reset
> 21:52:09.387 RSL: STATE = {"Time":"2021-08-09T21:52:09","Uptime":"0T00:00:09","UptimeSec":9,"Heap":29,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"xxxxx","BSSId":"xxxxxxxxx","Channel":6,"Mode":"11n","RSSI":100,"Signal":-42,"LinkCount":1,"Downtime":"0T00:00:03"}}

**3º** still with internet, add a simple relay to test lag / slow / unresponsiveness and reboot again: result is great: no lag at all, very fast indeed.
**4º** reboot and still with internet on, test if it lags; and no! still very fast, super fast, can't ask for more.
**5º** turn off internet, and it starts lagging, very slow, 5 to 10secs response time, even going around the menu, console, information, everything lags. and its not a network problem! already tried some other clean lan setups. and also disabled MQTT to test it.

> 00:00:00.001 HDW: ESP8266EX
> 00:00:00.050 CFG: Loaded from flash at F7, Count 17
> 00:00:00.056 QPC: Count 1
> 00:00:00.064 Project tasmota - Tasmota Version 9.5.0.5(tasmota)-2_7_4_9(2021-08-09T21:38:59)
> 00:00:00.066 I2C: DS3231 found at 0x68
> 00:00:00.076 Set time from DS3231 to RTC (UTC) 2021-08-09T21:00:13, (DST) 2021-03-28T02:00:00, (STD) 2021-10-31T03:00:00
> 00:00:00.504 WIF: Connecting to AP1 xxxxx Channel 6 BSSId xxxxxxxx in mode 11n as tasmota-xxxxx-xxxxx...
> 22:00:14.753 WIF: Connected
> 22:00:15.005 HTP: Web server active on tasmota-xxxx-xxxx with IP address 192.168.2.xx
> 22:00:15.008 RSL: INFO1 = {"Info1":{"Module":"Generic","Version":"9.5.0.5(tasmota)","FallbackTopic":"cmnd/DVES_389EE9_fb/","GroupTopic":"cmnd/tasmotas/"}}
> 22:00:15.017 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-xxxx-xxxxx","IPAddress":"192.168.2.xx"}}
> 22:00:15.028 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart"}}
> 22:00:15.035 RSL: RESULT = {"POWER":"OFF"}
> 22:00:15.038 RSL: POWER = OFF
> 22:00:29.465 QPC: Reset
> 22:00:29.470 RSL: STATE = {"Time":"2021-08-09T22:00:29","Uptime":"0T00:00:08","UptimeSec":8,"Heap":29,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"LAND","BSSId":"00:18:0A:F3:7D:E1","Channel":6,"Mode":"11n","RSSI":94,"Signal":-53,"LinkCount":1,"Downtime":"0T00:00:03"}}

any ideas what maybe? i don't think its a time related issue! with or without RTC, its the same. but! maybe i'm wrong

**Comment by s-hadinger** [comment]
Just to be sure, please try `SetOption127 1` and check if it still lags. 

I don't see anything in Tasmota depending on internet access for the web ui. Unfortunately I don't have a disconnected AP at hand. Something an esp32 can surely do. 

Your lagging sounds like reverse dns failing on the client side. Can you try a simple curl command to see if it lags as well?

**Comment by Jason2866** [comment]
What happens when you clear all NTP server entries?
```
NtpServer1 0
NtpServer2 0
NtpServer3 0
```

**Comment by arendst** [comment, marked as answer]
Or even better set one ntpserver to your gateway ip address and the others to "empty" like ``ntpserver2 "`` and ``ntpserver3 "``

You could also try the latest dev branch which has fixes regarding DNS lookup and NTP.
